features/steps/output.py

```diff
@@ -48,34 +50,55 @@ def then_not_in_conditional_stdout_does_not_match_regexp(
 def then_stream_does_not_match_regexp(context, stream):
     text = process_template_vars(context, context.text)
     content = getattr(context.process, stream).strip()
-    assert_that(content, not_(matches_regexp(text)))
+    if re.compile(text).search(content) is not None:
```

Comment:
Just a question on this - what do we win from moving away from hamcrest here? Is there a benefit not to rely on the functions we were using before?

Reply:
hamcrest gave errors that look like this:
```
      Assertion Failed: 
      Expected: a string matching 'SERVICE         +ENTITLED +STATUS   +DESCRIPTION
      cc-eal          +yes      +n/a      +Common Criteria EAL2 Provisioning Packages
      esm-apps        +yes      +enabled  +Expanded Security Maintenance for Applications
      esm-infra       +yes      +enabled  +Expanded Security Maintenance for Infrastructure
      fips            +yes      +n/a      +NIST-certified core packages
      fips-updates    +yes      +n/a      +NIST-certified core packages with priority security updates
      livepatch       +yes      +n/a      +Canonical Livepatch service
      realtime-kernel +yes      +n/a      +Ubuntu kernel with PREEMPT_RT patches integrated
      ros             +yes      +n/a      +Security Updates for the Robot Operating System
      ros-updates     +yes      +n/a      +All Updates for the Robot Operating System
      usg             +yes      +disabled +Security compliance and audit tools
      
      Enable services with: pro enable <service>'
           but: was 'SERVICE          ENTITLED  STATUS    DESCRIPTION\ncc-eal           yes                n/a                Common Criteria EAL2 Provisioning Packages\nesm-apps         yes                enabled            Expanded Security Maintenance for Applications\nesm-infra        yes                enabled            Expanded Security Maintenance for Infrastructure\nfips             yes                n/a                NIST-certified core packages\nfips-updates     yes                n/a                NIST-certified core packages with priority security updates\nlivepatch        yes                n/a                Canonical Livepatch service\nrealtime-kernel  yes                n/a                Ubuntu kernel with PREEMPT_RT patches integrated\n├ generic        yes                n/a                Generic version of the RT kernel (default)\n└ intel-iotg     yes                n/a                RT kernel optimized for Intel IOTG platform\nros              yes                n/a                Security Updates for the Robot Operating System\nros-updates      yes                n/a                All Updates for the Robot Operating System\nusg              yes                disabled           Security compliance and audit tools\n\nEnable services with: pro enable <service>\n\n                Account: UA Client Test\n           Subscription: server-team-ua-client-ci-uaa\n            Valid until: Fri Dec 31 00:00:00 9999 UTC\nTechnical support level: essential'
```
The problem I was facing is that it is hard to read that really long line at the bottom even when it's wrapped.

The new assertions here will give errors that look like this:
```
      Assertion Failed: Expected to match regexp:
        SERVICE         +ENTITLED +STATUS   +DESCRIPTION
        cc-eal          +yes      +n/a      +Common Criteria EAL2 Provisioning Packages
        esm-apps        +yes      +enabled  +Expanded Security Maintenance for Applications
        esm-infra       +yes      +enabled  +Expanded Security Maintenance for Infrastructure
        fips            +yes      +n/a      +NIST-certified core packages
        fips-updates    +yes      +n/a      +NIST-certified core packages with priority security updates
        livepatch       +yes      +n/a      +Canonical Livepatch service
        realtime-kernel +yes      +n/a      +Ubuntu kernel with PREEMPT_RT patches integrated
        ros             +yes      +n/a      +Security Updates for the Robot Operating System
        ros-updates     +yes      +n/a      +All Updates for the Robot Operating System
        usg             +yes      +disabled +Security compliance and audit tools

        Enable services with: pro enable <service>
      But got:
        SERVICE          ENTITLED  STATUS    DESCRIPTION
        cc-eal           yes                n/a                Common Criteria EAL2 Provisioning Packages
        esm-apps         yes                enabled            Expanded Security Maintenance for Applications
        esm-infra        yes                enabled            Expanded Security Maintenance for Infrastructure
        fips             yes                n/a                NIST-certified core packages
        fips-updates     yes                n/a                NIST-certified core packages with priority security updates
        livepatch        yes                n/a                Canonical Livepatch service
        realtime-kernel  yes                n/a                Ubuntu kernel with PREEMPT_RT patches integrated
        ├ generic        yes                n/a                Generic version of the RT kernel (default)
        └ intel-iotg     yes                n/a                RT kernel optimized for Intel IOTG platform
        ros              yes                n/a                Security Updates for the Robot Operating System
        ros-updates      yes                n/a                All Updates for the Robot Operating System
        usg              yes                disabled           Security compliance and audit tools

        Enable services with: pro enable <service>
```
Which I find a lot easier to spot the difference in

Reply:
love this